skrub/_datetime_encoder.py

```diff
@@ -185,9 +185,8 @@ def fit(self, X: ArrayLike, y=None) -> "DatetimeEncoder":
         if not hasattr(X, "__dataframe_consortium_standard__"):
             X = check_input(X)
             X = pd.DataFrame(X, columns=[str(i) for i in range(X.shape[1])])
-        X = X.__dataframe_consortium_standard__()
-        if hasattr(X, 'collect'):
-            X = X.collect()
+        X = X.__dataframe_consortium_standard__(api_version='2023.11-beta')
+        X = X.persist()
```

Comment:
What is the idea behind `persist()`?

Reply:
It makes sure that computations prior to this step won't be repeated https://data-apis.org/dataframe-api/draft/API_specification/dataframe_object.html#dataframe_api.DataFrame.persist . In the end, the consortium preferred this to `collect`. It does pretty much the same thing anyway (in Polars it does `.collect().lazy()`)

Below, `if df.col(col_name).std() > 0` will trigger computation because it calls `(df.col(col_name).std() > 0).__bool__()`. That command produces a Python scalar, so it's necessary to trigger compute here (as opposed to keeping things lazy)

By putting `persist`, we ensure that the compute which gets triggered only happens from this point until the `__bool__` call

Just saw you've already done a lot of work here to support Polars - well done! I'll rebase then